sdk/core/Azure.Core.Experimental/src/DynamicJson.cs

```diff
@@ -38,7 +40,12 @@ internal override void WriteTo(Stream stream)
 
         private object GetProperty(string name)
         {
-            return new DynamicJson(_element.GetProperty(name));
+            if (_element.TryGetProperty(name, out MutableJsonElement element))
+            {
+                return new DynamicJson(element);
+            }
+
+            return new DynamicJson(MutableJsonDocument.Parse(nullJsonData).RootElement);
```

Comment:
We have to parse this as a new document so that callers an update the value?

Reply:
We do, and actually staring at this, I think there may be a bug here.  Will check and update.

Reply:
Changed the approach and now return a property `null` instead of DynamicJson wrapping a null JsonValueKind.